packages/typespec-ts/src/utils/modelUtils.ts

```diff
@@ -1197,6 +1197,30 @@ export function getTypeName(schema: Schema, usage?: SchemaContext[]): string {
   return getPriorityName(schema, usage) ?? schema.type ?? "any";
 }
 
+export function getSerializeTypeName(
+  schema: Schema,
+  usage?: SchemaContext[]
+): string {
+  const typeName = getTypeName(schema, usage);
+  const formattedName = typeName.replace("Date | string", "string");
+  const allTypes = formattedName.includes(" | ")
+    ? formattedName.split(" | ")
+    : formattedName.split("|");
+  const canSerialize = allTypes.every((type) => {
+    return (
+      ["string", "number", "boolean"].includes(type) ||
+      (type.startsWith('"') && type.endsWith('"')) ||
+      /^[0-9]*$/.test(type) ||
+      ["true", "false"].includes(type) ||
+      ["null"].includes(type)
```

Comment:
I remember currently the null value is not allowed in core for headers.

Reply:
https://github.com/Azure/autorest.typescript/blob/main/packages/typespec-test/test/widget_dpg/generated/typespec-ts/src/rest/parameters.ts#L18 I may have lost some context, is this not working ?